### ISSUE TITLE
Update customer callback logic and implementation

### ISSUE THREAD

**Issue by Ifropc** [stellar/java-stellar-anchor-sdk#1475]
### Background information

Currently, there are a few flaws with customer callback logic:
First,
Business server must always call `notify_customer_info_updated` on status update, even if the original request comes from the Anchor Platform, as part of `PUT` callback. If business doesn't call RPC, transaction status [is not updated appropriately](https://github.com/philipliu/java-stellar-anchor-sdk/blob/2e90acc85f770e4b91ff985ae939a20a20c842df/core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java#L109-L125).
We make a callback to wallet both in RPC and in `PUT /customer` that leads to duplicate update on the wallet size.
Second, according to the SEP-12 spec, we should only send callbacks when status has been changed. Currently, we send callback on any `PUT` request and any RPC call (that is also done as part of `PUT` request handling).

### Business server `PUT` callback proposal change
We propose to change `PUT` callback for the business server. It should now return full Customer body response. This will allow us to not require business to call RPC on sync updates anymore. Instead, we can perform the same logic in `PUT` handler (update transaction status) 
Second, we should add a new required boolean field to Business Server `PUT` callback response (`status_updated`). When set to `true`, we will fire an customer status change event. If set to `false`, nothing happens. 
In return, as RPC is used to sync updates, we should always fire customer status change events (the way it's done now)

**Comment by lijamie98** [comment]
> Second, we should add a new required boolean field to Business Server PUT callback response (status_updated). When set to true, we will fire an customer status change event. If set to false, nothing happens.

Why do we need the flag? If the AP sets the flag to `true`, AP already know if it should fire the status change event. If `false`, the AP also know it will not fire the event.

**Comment by lijamie98** [comment]
> We propose to change PUT callback for the business server. It should now return full Customer body response. This will allow us to not require business to call RPC on sync updates anymore. Instead, we can perform the same logic in PUT handler (update transaction status)

In the sync case, what do we do if the BS calls the RPC?

**Comment by Ifropc** [comment]
> In the sync case, what do we do if the BS calls the RPC? Can we elaborate?

We don't need to do anything, it's not a problem if BS calls the RPC. We just note in docs, that BS should not call RPC in the sync case (if they do, it will just create a duplicate event, that's it)

**Comment by Ifropc** [comment]
> Why do we need the flag? If the AP sets the flag to true, AP already know if it should fire the status change event. If false, the AP also know it will not fire the event.

It's in the response of business server, we don't know the value of it ahead of time

**Comment by lijamie98** [comment]
>  Instead, we can perform the same logic in PUT handler (update transaction status)
I don't think this is a good idea because updating the transaction should be handled via RPC.

I propose that we only publish the customer callback in the `PUT /customer` endpoint of `Sep12Service` when the `status_updated` is `true`. The `notify_customer_info_updated` RPC handler should not trigger a callback event and only handle transaction status update.

This should prevent duplicated events.

**Comment by JiahuiWho** [comment]
I was even thinking maybe we can save the `status_updated`  flag

Yesterday we conclude that we need to 
1 In SEP-12, add `status_updated`  flag to customer callback response to decide whether we should send a call back to wallet
2. In RPC, remove the event firing in the `NOTIFY_CUSTOMER_INFO_UPDATED` 

But with 2 implemented, I think it's ok to not have 1. 
The result would be we always send a callback (with the latest customer info regardless updated or not) to wallet when there's a put request, regardless it's info being updated or not.

I believe consistent callback behavior is important for developers. With predictable callbacks, developers understand that this is the expected successful response, making it easy to verify that the PUT request was successful and that all submitted information is correct. However, if callbacks are intermittent, it can create confusion about whether the request succeeded, requiring an additional GET call to confirm.
